src/Robo/Plugin/Commands/DevelopmentModeBaseCommands.php

```diff
@@ -123,43 +123,57 @@ public function databaseRefreshLando(string $siteName = 'default'): Result
     /**
      * Refresh database on Tugboat.
      */
-    public function databaseRefreshTugboat(): Result
+    public function databaseRefreshTugboat(): ResultData
     {
         $this->io()->title('refresh tugboat databases.');
-        $result = null;
+        $resultData = new ResultData();
+
         foreach (array_keys($this->getAllSitesConfig()) as $siteName) {
+            $dbPath = '';
             try {
                 $dbPath = $this->databaseDownload($siteName);
-            } catch (TaskException) {
+            } catch (TaskException $e) {
                 $this->yell("$siteName: No database configured. Download/import skipped.");
+                $resultData->append($e->getMessage());
                 // @todo: Should we run a site-install by default?
                 continue;
             }
-            if (!is_string($dbPath) || strlen($dbPath) == 0) {
+            if (!is_string($dbPath) || strlen($dbPath) === 0) {
```

Comment:
Ok. So I can get around the fact that phpstan doesn't like `empty($dbPath)` and phpstan doesn't like `isset()` on a variable that (it thinks) is defined, though it is defined in a try and won't be defined in the case of a failure, by forcing the variable to be set to an empty string at top. PHPStorm doesn't like that either, but I can tell PHPStorm to be quiet. 

But now, Rector is complaining that my test for the empty string should not use `strlen()` but should use `empty()` and we know how phpstan feels about `empty()`

So phpstan hates `empty()` because it is vague, and Rector hates `strlen($var) === 0` and wants me to use `empty()` .... 

Making me think maybe I have to do a string to string comparison like `$dbPath === ''` which, I guess is ok, but feels wrong to me. 

so... spinning my wheels between these validators complaining.

Reply:
Ok. I'm just going to go with `$dbPath === ''`  I have this "sensation" as a developer that comparing a string to '' is somehow more error prone than strlen() === 0 or empty(). But I don't know why. 🤔

Reply:
Sounds good!